pennylane/ops/op_math/exp.py

```diff
@@ -277,9 +276,21 @@ def _recursive_decomposition(self, base: Operator, coeff: complex):
             ops = base.ops if isinstance(base, Hamiltonian) else base.operands
             return self._trotter_decomposition(ops, coeffs)
 
-        for op_class in has_unitary_generator_types:
-            # Check if the exponentiated operator is a generator of another operator
-            if op_class.num_wires in {base.num_wires, AnyWires} and op_class is not qml.PauliRot:
+        # Store operator classes with generators
+        has_generator_types = []
+        has_generator_types_anywires = []
+        for op_name in qml.ops.qubit.__all__:  # pylint:disable=no-member
+            op_class = getattr(qml.ops.qubit, op_name)  # pylint:disable=no-member
+            if op_class.has_generator and op_class.num_wires == AnyWires:
+                has_generator_types_anywires.append(op_class)
+            elif op_class.has_generator and op_class.num_wires == base.num_wires:
+                has_generator_types.append(op_class)
```

Comment:
```suggestion
            if op_class.has_generator:
                if op_class.num_wires == AnyWires:
                    has_generator_types_anywires.append(op_class)
                elif op_class.num_wires == base.num_wires:
                    has_generator_types.append(op_class)
```

Since `has_generator` is on both of these branches, best to extract it.